examples/on_saved/src/benchmark.py

```diff
@@ -0,0 +1,223 @@
+import tempfile
```

Comment:
Can you rename that directory to something more expressive, like "examples/artifact-benchmarking"?